app/models/lookup_keys/variable_lookup_key.rb

```diff
@@ -2,7 +2,7 @@ class VariableLookupKey < LookupKey
   belongs_to :puppetclass, :inverse_of => :lookup_keys, :counter_cache => :variable_lookup_keys_count
 
   validates :puppetclass, :presence => true
-  validates :key, :uniqueness => true
+  validates :key, :uniqueness => true, :no_whitespace => true
```

Comment:
I think it should be in the parent since it isn't specific to VariableLKs but rather a property of any variable name, even if the PuppetLKs are supposed to take care of it during import.